src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/WaitingListService.kt

```
@@ -56,7 +57,17 @@ class WaitingListService(
     scheduleRepository
       .findOrThrowNotFound(id)
       .checkCaseloadAccess()
-      .let(waitingListRepository::findByActivitySchedule).map(WaitingList::toModel)
+      .let(waitingListRepository::findByActivitySchedule).map(WaitingList::toModel).apply {
+        val prisoners =
+          prisonerSearchApiClient.findByPrisonerNumbers(map { it.prisonerNumber })
+
+        map {
+          val prisoner = prisoners.find { p -> it.prisonerNumber == p.prisonerNumber }
+          if (prisoner != null) {
```

Hmm `prisoner == null` should never happen.

If it does then we should probably fail fast here, something is wrong with our data.

Doing this though obviously has ramifications as retrieval of waiting lists will fail also.

---

Does that mean the test I edited below (WaitingListServiceTest ) the line below could never happen? I should always return a value? If I added a return object it wouldn't need a null check. 
 
prisonerSearchApiClient.stub {
      on { findByPrisonerNumbers(listOf("123456")) } doReturn emptyList()
    }

The earliestReleaseDate is optional in the schema